src/functional/merge.js

```diff
@@ -69,10 +72,19 @@ export function mergeWithSources(collection, sources, merger) {
 function deepMergerWith(merger) {
   function deepMerger(oldValue, newValue, key) {
     return isDataStructure(oldValue) && isDataStructure(newValue)
-      ? mergeWithSources(oldValue, [newValue], deepMerger)
+      ? areMergeable(oldValue, newValue)
+        ? mergeWithSources(oldValue, [newValue], deepMerger)
+        : newValue
       : merger
       ? merger(oldValue, newValue, key)
       : newValue;
   }
   return deepMerger;
 }
+
+function areMergeable(oldValue, newValue) {
+  return !(
+    (isIndexed(oldValue) || Array.isArray(oldValue)) &&
```

Comment:
Hmm, I'm open to that. The reason I kept it so narrow is because these checks match the checks in `isDataStructure` which is what is used in `deepMergerWith`. Do you still think it's worth updating?

Reply:
I have low confidence, but that was my instinct. Maybe something you can confirm with test coverage.

Reply:
Done.